### ISSUE TITLE
Raw types

### ISSUE THREAD

**Issue by davidmoten** [microsoftgraph/msgraph-sdk-java#37]
As a cleanup issue, there is a lot of raw type usage in the project. Quoting from the Java Language Specification:

The use of raw types is allowed only as a concession to compatibility of legacy code. The use of raw types in code written after the introduction of generics into the Java programming language is strongly discouraged. It is possible that future versions of the Java programming language will disallow the use of raw types.
[AB#6043](https://microsoftgraph.visualstudio.com/0985d294-5762-4bc2-a565-161ef349ca3e/_workitems/edit/6043)

**Comment by deepak2016** [comment]
@davidmoten given that changes for this one has been merged, can I go ahead and close this issue?

**Comment by davidmoten** [comment]
Have you checked for more raw type usage? In generated code?

**Comment by deepak2016** [comment]
I don't see any warning getting emitted for generated code. I see two instances of raw type usages.
1. ChunkedUploadResult is a raw type. References to generic type ChunkedUploadResult<UploadType> should be parameterized	ChunkedUploadResponseHandler.java	/msgraph-sdk-java/src/main/java/com/microsoft/graph/concurrency	line 45	Java Problem

2. Iterator is a raw type. References to generic type Iterator<E> should be parameterized	EnumSetSerializer.java	/msgraph-sdk-java/src/main/java/com/microsoft/graph/serializer	line 67	Java Problem

Fixing second one is straightforward, I need more understanding to fix the first issue.

**Comment by davidmoten** [comment]
Re 1., the signature is:

```java
public class ChunkedUploadResponseHandler<UploadType>
        implements IStatefulResponseHandler<ChunkedUploadResult, UploadType>
```
That should presumably be changed to:

```java
public class ChunkedUploadResponseHandler<UploadType>
        implements IStatefulResponseHandler<ChunkedUploadResult<UploadType>, UploadType>
```
I'm not sure about that last `UploadType` on the signature, looks a bit odd because it corresponds to a return type so I'd expect perhaps `Void` but not a repeat of `UploadType`. Keep digging and let me know if you can't sort it out.

**Comment by baywet** [comment]
closing as fixed by #115
